core/txpool/legacypool/legacypool.go

```diff
@@ -827,7 +881,23 @@ func (pool *LegacyPool) add(tx *types.Transaction, local bool) (replaced bool, e
 					pool.priced.Put(dropTx, false)
 				}
 				log.Trace("Discarding future transaction replacing pending tx", "hash", hash)
-				return false, txpool.ErrFutureReplacePending
+				return false, txpool.ErrFutureReplacePending // todo 1 maybe in this case the future transaction can be part of pool3?
+			}
+		}
+
+		// calculate total number of slots in drop. Accordingly add them to pool3 (if there is space)
+		// all members of drop will be dropped from pool1/2 regardless of whether they get added to pool3 or not
+		availableSlotsPool3 := pool.availableSlotsPool3()
```

Comment:
Can the call here be simplified? Let the complex logic be coupled in the pool3 object? For example, for pool3, which is an LRU, just add it directly and let the LRU eliminate it.
```go
pool.addToPool3(drop, isLocal)
```